<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1024</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>OpenEMSH</string>
  </property>
  <property name="windowState">
   <enum>Qt::WindowMaximized</enum>
  </property>
  <property name="dockNestingEnabled">
   <bool>true</bool>
  </property>

  <widget class="QDockWidget" name="dw_processing">
   <property name="windowTitle">
    <string>Processing flow</string>
   </property>
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <attribute name="dockWidgetArea">
    <number>4</number>
    <!--<enum>Qt::TopDockWidgetArea</enum>-->
   </attribute>
   <widget class="QWidget" name="w_processing">
    <layout class="QHBoxLayout" name="hbl_processing">
     <item>
      <layout class="QVBoxLayout" name="vbl_processing_controls">
       <item>
        <widget class="QToolButton" name="tb_snap">
         <property name="text">
          <string>∷</string>
         </property>
         <property name="toolTip">
          <string>Snap items</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="tb_show_selected">
         <property name="text">
          <string>∼</string>
         </property>
         <property name="toolTip">
          <string>Show selected chain only</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="tb_show_displayed">
         <property name="text">
          <string>≈</string>
         </property>
         <property name="toolTip">
          <string>Show axes/plane currently displayed only</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="tb_show_everything">
         <property name="text">
          <string>≋</string>
         </property>
         <property name="toolTip">
          <string>Show everything</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="tb_curved_wires">
         <property name="text">
          <string>↝</string>
         </property>
         <property name="toolTip">
          <string>Curved wires</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="tb_direct_wires">
         <property name="text">
          <string>↗</string>
         </property>
         <property name="toolTip">
          <string>Direct wires</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="s_processing_controls">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QGraphicsView" name="processing_view">
       <property name="sizeAdjustPolicy">
        <enum>QAbstractScrollArea::AdjustToContentsOnFirstShow</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSlider" name="s_processing_zoom">
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>500</number>
       </property>
       <property name="value">
        <number>250</number>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>

  <widget class="QDockWidget" name="dw_structure">
   <property name="windowTitle">
    <string>Structure topology</string>
   </property>
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
    <!--<enum>Qt::BottomDockWidgetArea</enum>-->
   </attribute>
   <widget class="QWidget" name="w_structure">
    <layout class="QHBoxLayout" name="hbl_structure">
     <item>
      <layout class="QVBoxLayout" name="vbl_structure_controls">
       <item>
        <widget class="QToolButton" name="tb_anchor">
         <property name="text">
          <string>⨁</string>
         </property>
         <property name="toolTip">
          <string>Centered view anchor</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="tb_reset">
         <property name="text">
          <string>⊹</string>
         </property>
         <property name="toolTip">
          <string>Reset view</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="tb_vertical_layout">
         <property name="text">
          <string>┃</string>
         </property>
         <property name="toolTip">
          <string>Vertical layout</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="tb_horizontal_layout">
         <property name="text">
          <string>━</string>
         </property>
         <property name="toolTip">
          <string>Horizontal layout</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="tb_show_all_mesh">
         <property name="text">
          <string>▦</string>
         </property>
         <property name="toolTip">
          <string>Show mesh</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="tb_show_vertical_mesh">
         <property name="text">
          <string>▥</string>
         </property>
         <property name="toolTip">
          <string>Show vertical meshlines only</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="tb_show_horizontal_mesh">
         <property name="text">
          <string>▤</string>
         </property>
         <property name="toolTip">
          <string>Show horizontal meshlines only</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="tb_show_no_mesh">
         <property name="text">
          <string>⊡</string>
         </property>
         <property name="toolTip">
          <string>Don't show mesh</string>
         </property>
        </widget>
       </item>

       <item>
        <widget class="QRadioButton" name="rb_plane_yz">
         <property name="text">
          <string>YZ</string>
         </property>
         <property name="toolTip">
          <string>View YZ plane</string>
         </property>
         <property name="focusPolicy">
          <enum>Qt::NoFocus</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="rb_plane_zx">
         <property name="text">
          <string>ZX</string>
         </property>
         <property name="toolTip">
          <string>View ZX plane</string>
         </property>
         <property name="focusPolicy">
          <enum>Qt::NoFocus</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="rb_plane_xy">
         <property name="text">
          <string>XY</string>
         </property>
         <property name="toolTip">
          <string>View XY plane</string>
         </property>
         <property name="focusPolicy">
          <enum>Qt::NoFocus</enum>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="s_structure_controls">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
     <item>
      <widget class="ui::qt::StructureView" name="structure_view">
       <property name="sizeAdjustPolicy">
        <enum>QAbstractScrollArea::AdjustToContentsOnFirstShow</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSlider" name="s_structure_zoom">
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>500</number>
       </property>
       <property name="value">
        <number>250</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSlider" name="s_structure_rotation">
       <property name="minimum">
        <number>-180</number>
       </property>
       <property name="maximum">
        <number>180</number>
       </property>
       <property name="value">
        <number>0</number>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>

  <widget class="QMenuBar" name="menuBar">
   <addaction name="a_about"/>
  </widget>
  <action name="a_about">
   <property name="text">
    <string>About</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ui::qt::StructureView</class>
   <extends>QGraphicsView</extends>
   <header>ui/qt/structure_view/structure_view.hpp</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
